src/raqm.c

```diff
@@ -377,14 +399,38 @@ raqm_destroy (raqm_t *rq)
   if (!rq || --rq->ref_count != 0)
     return;
 
-  free (rq->text);
-  free (rq->text_utf8);
-  _raqm_free_text_info (rq);
+  _raqm_release_text_info (rq);
+  _raqm_free_text (rq);
   _raqm_free_runs (rq);
   free (rq->glyphs);
+  free (rq->features);
   free (rq);
 }
 
+/**
+ * raqm_clear_contents:
+ * @rq: a #raqm_t.
+ *
+ * Clears internal state of previously used raqm_t object, making it ready
+ * for reuse and keeping some of allocated memory to increase performance.
+ *
+ * Since: 0.9
+ */
+void
+raqm_clear_contents (raqm_t *rq)
+{
+  if (!rq)
+    return;
+
+  _raqm_release_text_info (rq);
+
+  _raqm_free_runs (rq);
+  rq->runs = NULL;
```

Comment:
It would be nice if we could re-use the runs (`hb_buffer_t` can be reused), but I think this might require larger changes.

Reply:
I wanted to do this initially but since it is a linked list there needs to be a pool of nodes, which could quickly became too large for me to do right now. You may consider this as a further improvement.

Reply:
Yes, that is the larger change I had in mind.